test/helper/test_trigger.ts

```diff
@@ -586,7 +591,60 @@ const testTriggerFile = (file: string) => {
 
       const triggers = triggerSet.triggers;
       for (const trigger of triggers ?? []) {
-        const origRegex = trigger.netRegex?.source.toLowerCase();
+        if (trigger.netRegex === undefined)
+          continue;
+
+        if (trigger.type === undefined) {
+          if (!(trigger.netRegex instanceof RegExp)) {
+            assert.fail(
+              `${trigger.id} doesn't have 'type' property and doesn't have a RegExp netRegex`,
+            );
+          }
+          continue;
+        }
+
+        if (!(trigger.netRegex instanceof RegExp)) {
+          // plain object netRegex
+          if (trigger.disabled)
+            continue;
+
+          const textHasTranslation = (text: string): boolean => {
+            return translateWithReplacements(
+              text,
+              'replaceSync',
+              locale,
+              translations,
+            ).wasTranslated;
+          };
+
+          const fieldHasTranslation = (field: string | string[], fieldName: string) => {
```

Comment:
Can this be named `checkIfFieldHasTranslation` so it has a verb in the name? There's `textHasTranslation` which returns a boolean if the text was translated, and `fieldHasTranslation` which performs some assert checks and doesn't return anything.  I think having them both named so similarly is confusing.